pkg/promlib/admission_handler.go

```diff
@@ -0,0 +1,77 @@
+package promlib
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var (
+	_ backend.AdmissionHandler = (*Service)(nil)
+)
+
+// ValidateAdmission implements backend.AdmissionHandler.
+func (s *Service) ValidateAdmission(ctx context.Context, req *backend.AdmissionRequest) (*backend.ValidationResponse, error) {
+	rsp, err := s.MutateAdmission(ctx, req)
+	if rsp != nil {
+		return &backend.ValidationResponse{
+			Allowed:  rsp.Allowed,
+			Result:   rsp.Result,
+			Warnings: rsp.Warnings,
+		}, err
+	}
+	return nil, err
+}
+
+// MutateAdmission implements backend.AdmissionHandler.
+func (s *Service) MutateAdmission(ctx context.Context, req *backend.AdmissionRequest) (*backend.MutationResponse, error) {
+	expected := (&backend.DataSourceInstanceSettings{}).GVK()
```

Comment:
This is a bit strange - is this a long term thing?

Reply:
probably not, the goal here is to check that we are receiving a `DatasourceInstanceSettings`, which could be moved to a middleware or something so people don't need to do this for every plugin.